packages/flutter_tools/lib/src/test/flutter_web_platform.dart

```diff
@@ -193,6 +200,38 @@ class FlutterWebPlatform extends PlatformPlugin {
     }
   }
 
+  final shelf.Handler _flutterFilesStaticHandler = createStaticHandler(
+        globals.fs.path.join(Cache.flutterRoot, 'packages', 'flutter', 'lib', 'src'),
+        serveFilesOutsidePath: true,
+      );
+
+  FutureOr<shelf.Response> _flutterFilesHandler(shelf.Request request) {
+    // For some reason, the paths we are getting are missing the `lib/` segment.
+    // So we remove the "packages/flutter/src/" prefix and serve it as a relative
+    // path from inside the flutter src directory.
+    if (request.requestedUri.path.startsWith('/packages/flutter/src')) {
```

Comment:
Is this the correct strategy for sourcemap files that aren't in `package:flutter`? I would expect something like `package/hello_world/main.dart` for example.  I think what you want to do is load the .packages file, that will give you a map from package name -> file uri. Then you can join the fragment at `packages/<package-name>/<rest>` to find the correct file path.